### ISSUE TITLE
Negative Wastage

### ISSUE THREAD

**Issue by fauder** [ThomasLoke/stock_cutter#3]
Hello,

First, thanks for authoring this and including an MIT license for others to use. It's been immensly useful in my project.

Now on to the problem I have:
Stock-Cutter finds an optimal solution in which there is negative wastage.
 
For example, for a stock length of 6000, stock-cutter gave the pattern 1550  2300  2850, which amount to 6700 when summed. There is -700 wastage. 

Is there a way to make it so that there never is negative wastage ?

**Comment by ThomasLoke** [comment]
Hi @fauder, could you upload the input CSV file that illustrates this problem?

**Comment by fauder** [comment]
Okay I think I understand what's wrong. It's not exactly as I described. Stock-Cutter generates this:
```
Order quantity supplied, 6, 3, 7, 1, 1, 2, 5, 2
Order quantity required, 6, 3, 7, 1, 1, 2, 5, 1
```
I put patterns in a multimap indexed by length and use std::lower_bound() to obtain, use and remove them from the map. lower_bound() tries to find a 2nd instance of that last order length but since it isn't there it finds the next closest I think.
i.e. It tries to find a second 1550 but there is only one 1550, so it finds the next length 2300 and uses that.

All in all nothing wrong with Stock-Cutter. I got to fix this on my end.
